packages/host/app/services/matrix-service.ts

```diff
@@ -239,11 +239,15 @@ export default class MatrixService extends Service {
     cardController.workspaceChooserOpened = true;
     this.start(auth);
     this.setDisplayName(displayName);
+    await this.initializeNewUserRealm.perform(displayName);
+    await this.loginToRealms();
+  }
 
+  initializeNewUserRealm = task(async (realmName: string) => {
```

Comment:
I moved the process of initializing a new user's realm into a task. This way, we can check the processing status and display the WorkspaceLoadingIndicator in the workspace chooser even when there are no realms to display yet. (I'm open to other approaches).

Reply:
i don't see how having this be a task makes any difference. this method lives in a service which is long lived. A task is really meant for owners that have short life-times like components. That way the task can be cancelled safely when the owner goes out of scope. a service never goes out of scope so there really is no reason to need a cancellable promise. can you explain more clearly why having a task is desirable here?